packages/perseus-score/src/validate.ts

```diff
@@ -36,3 +36,34 @@ export function emptyWidgetsFunctional(
         return false;
     });
 }
+
+// TODO: this is very similar to emptyWidgetsFunctional,
+// they should be merged
+export function validateWidgets(
+    widgets: ValidationDataMap,
+    // This is a port of old code, I'm not sure why
+    // we need widgetIds vs the keys of the widgets object
+    widgetIds: ReadonlyArray<string>,
```

Comment:
I _think_ this list of widget IDs is extracted from the `content` string and thus guaranteed to be in use, whereas the `widgets` map could include unused widgets. 

I _wonder_ if our parsing system should parse the `content` string and strip any widgets in the `widgets` dictionary that aren't in use.